### ISSUE TITLE
Upload: use Servlet 3 API when supported

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#996]
See #985 for background information.

The idea here is that the container has the ability to close the connection. But this has to be verified, at least with Tomcat.

A quick glance at Tomcat sources seems to indicate that a file upload, when [`swallowAbortedUploads`](http://tomcat.apache.org/tomcat-7.0-doc/config/context.html) is set to `false` (which is _not_ the default!), will promptly close the connection. See [`Request.java`](https://github.com/apache/tomcat/blob/7f616f97067b8e3461f8611f9d26f0b262c4351f/java/org/apache/catalina/connector/Request.java#L768) as a starting point.

In short:
- using Tomcat 7
- with `swallowAbortedUploads` is set to `false`
- and Orbeon Forms using the Servlet 3 upload API

would be an improvement over the current situation, where the connection cannot be closed.

Drawbacks:
- support for pre-Servlet 3 containers
- does not resolve handling non multipart large request bodies (like a POST to a service)

**Comment by ebruchez** [comment]
Servlet 3.0 API is no longer an issue as it is now many years old and widely supported.
